### PR TITLE
docs: add changelog

### PR DESCRIPTION
**Pull request by jezekra1**
Add changelog generator

### CODE REVIEW

File: scripts/docs_changelog_generator/config.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      .
+    )*
+)
+
+(?P<rev>%(rev)s)
+""" % {"rev": r"v[0-9]+\.[0-9]+(\.[0-9]+)?"}
+
+section_regexps = [
+    ("New", [r"^[nN]ew\s*:\s*((dev|use?r|pkg|test|doc)\s*:\s*)?([^\n]*)$"]),
+    ("Changes", [r"^[cC]hg\s*:\s*((dev|use?r|pkg|test|doc)\s*:\s*)?([^\n]*)$"]),
```
Comment (Tomas2D) [inline scripts/docs_changelog_generator/config.py:43]: Add some emojis (even duplicated).

File: scripts/docs_changelog_generator/generate_changelog.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+from docs_changelog_generator.utils import GithubRSTLinks, get_git_tags, output_engine_pipeline, shave_marks
+from genai._generated.endpoints import ApiEndpoint
+
+load_dotenv()
+
+logger = get_logger(__name__)
+DIRNAME = Path(__file__).parent.absolute()
+GITCHANGELOG_CONFIG_FILENAME = str(DIRNAME / Path("config.py"))
+AUTHOR_NAMES_MAPPING_PATH = Path(__file__).parent.absolute() / Path("authors_by_name.yaml")
```
Comment (Tomas2D) [inline scripts/docs_changelog_generator/generate_changelog.py:19]: `DIRNAME / Path("authors_by_name.yaml")`

File: scripts/docs_changelog_generator/parse_releases.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        body_pre_convert = release["body"]
+        # Remove "What's changed" header
+        body_pre_convert = re.sub(r"#+\s*What'?s? [Cc]hanged\s*\n", "", body_pre_convert)
+
+        # Convert MD to RST
+        body_converted = m2r.convert(body_pre_convert, renderer=MyRenderer())
+
+        # Convert compare links
+        body_converted = re.sub(
+            r"https://github.com/IBM/ibm-generative-ai/compare/v([0-9]\.[0-9]\.[0-9])...v([0-9]\.[0-9]\.[0-9])",
```
Comment (Tomas2D) [inline scripts/docs_changelog_generator/parse_releases.py:35]: I would extract these links / names to constants.

File: scripts/docs_changelog_generator/parse_releases.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    for release in releases:
+        name = (
+            release["name"].replace("Release ", "")
+            + f" ({datetime.fromisoformat(release['published_at']).strftime('%Y-%m-%d')})"
+        )
+        result += [name]
+        result += ["-" * len(name)]
+
+        class MyRenderer(m2r.RestRenderer):
+            hmarks = {1: "^", 2: "^", 3: "^", 4: "^", 5: "^", 6: "^"}
```
Comment (Tomas2D) [inline scripts/docs_changelog_generator/parse_releases.py:24]: add comment

File: scripts/docs_changelog_generator/release_changelog.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+parser = argparse.ArgumentParser()
+parser.add_argument("version", help="New version tag to release")
+parser.add_argument(
+    "--date",
+    required=False,
+    type=datetime.fromisoformat,
+    help="Optional date of the release YYYY-MM-DD, today by default",
+    default=dateutil.utils.today(),
+)
+args = parser.parse_args()
```
Comment (Tomas2D) [inline scripts/docs_changelog_generator/release_changelog.py:25]: I would like to see this in a it's own function.

File: scripts/docs_changelog_generator/utils.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import unicodedata
+from pathlib import Path
+
+from gitchangelog.gitchangelog import GitCommit, GitRepos
+
+DEFAULT_REPO_URL = "https://github.com/IBM/ibm-generative-ai"
```
Comment (Tomas2D) [inline scripts/docs_changelog_generator/utils.py:6]: re-use on other places

File: scripts/docs_changelog_generator/utils.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+
+def output_engine_pipeline(render_fn):
+    def _process(output_engine):
+        return lambda data, opts: output_engine(*render_fn(data, opts))
+
+    return _process
```
Comment (Tomas2D) [inline scripts/docs_changelog_generator/utils.py:13]:
```python
output_engine_pipeline = lambda render_fn: lambda output_engine: lambda data, opts: output_engine(*render_fn(data, opts))
```

😄

**Review by Tomas2D — CHANGES_REQUESTED** [review]
I would extract URLs / Configuration values to some general config. 
Otherwise 👍🏻

**Review by Tomas2D — COMMENTED** [review]
Can we add anchors for each version to use links to versions inside the RST files?

```
.. _version_2.0.0:
```

File: scripts/docs_changelog_generator/utils.py (hunk 7 above)
Comment (jezekra1) [reply scripts/docs_changelog_generator/utils.py]: noooo 🙈 :D 
I add more code just to prevent this :D

**Review by Tomas2D — APPROVED** [review]

**Pull request merged**